### PR TITLE
add invoke element in java controller

### PR DESCRIPTION
**Pull request by yunhanw-google**
https://github.com/project-chip/connectedhomeip/issues/24931

--Add invoke element in java controller

### CODE REVIEW

File: src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java [hunk 1]
```diff
@@ -23,9 +23,9 @@
 
 /** An attribute write request that should be used for interaction model write interaction. */
 public final class AttributeWriteRequest {
-  private final ChipPathId endpointId, clusterId, attributeId;
-  private final Optional<Integer> dataVersion;
-  private final byte[] tlv;
+  private ChipPathId endpointId, clusterId, attributeId;
+  private Optional<Integer> dataVersion;
+  private byte[] tlv;
```
Comment (g-coppock) [inline src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java:28]: Was this change accidental? The fields aren't mutated.

File: src/controller/java/src/chip/devicecontroller/model/InvokeRequest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+package chip.devicecontroller.model;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+/** An invoke request that should be used for interaction model invoke command. */
+public final class InvokeRequest {
+  private ChipPathId endpointId, clusterId, commandId;
+  private byte[] tlv;
```
Comment (g-coppock) [inline src/controller/java/src/chip/devicecontroller/model/InvokeRequest.java:27]: Fields can be `final` here.

**Review by g-coppock — APPROVED** [review]

**Comment by yufengwangca** [comment]
LGTM, provided kevin's left comments are addressed

**Review by yufengwangca — APPROVED** [review]

File: src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java (hunk 1 above)
Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java:28]: yes, accidental copy from the earlier stage of write pr...fixed, thanks

File: src/controller/java/src/chip/devicecontroller/model/InvokeRequest.java (hunk 2 above)
Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/model/InvokeRequest.java:27]: done, thanks

File: src/controller/java/src/chip/devicecontroller/model/InvokeElement.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    return Objects.hash(endpointId, clusterId, commandId);
+  }
+
+  @Override
+  public String toString() {
+    return String.format(
+        Locale.ENGLISH, "Endpoint %s, cluster %s, command %s", endpointId, clusterId, commandId);
+  }
+
+  public static InvokeElement newInstance(
```
Comment (cliffamzn) [inline src/controller/java/src/chip/devicecontroller/model/InvokeElement.java:75]: nit, a javadoc might be nice here and below as well

**Review by cliffamzn — APPROVED** [review]

File: src/controller/java/src/chip/devicecontroller/model/InvokeElement.java (hunk 3 above)
Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/model/InvokeElement.java:75]: sure, would add it in https://github.com/project-chip/connectedhomeip/issues/25473, thanks

**Pull request merged**